activerecord/lib/active_record/persistence.rb

```diff
@@ -1265,12 +1265,22 @@ def verify_readonly_attribute(name)
 
     def _raise_record_not_destroyed
       @_association_destroy_exception ||= nil
-      key = self.class.primary_key
-      raise @_association_destroy_exception || RecordNotDestroyed.new("Failed to destroy #{self.class} with #{key}=#{send(key)}", self)
+      raise @_association_destroy_exception || RecordNotDestroyed.new(record_not_destroyed_i18n, self)
     ensure
       @_association_destroy_exception = nil
     end
 
+    def record_not_destroyed_i18n
```

Comment:
What do you think about moving this logic into `ActiveRecord::RecordNotDestroyed` instead?

Reply:
Yes, this should be in the exception like it is the RecordInvalid https://github.com/rails/rails/blob/c05f16e03ab27c13800da7df077bccc121d1e36c/activerecord/lib/active_record/validations.rb#L22